src/components/containers/ChartContainer.js

```diff
@@ -17,8 +18,71 @@ export default class extends React.Component {
 
             this.dataPack = this._createDataPack(props.unformattedData, this.chartType);
         }
+
+        this.annotations = [];
+
+        let dataToShow = {};
+        if (this.dataPack.trimmed && props.showOutliers === false) {
+            dataToShow = this.dataPack.trimmed;
+        } else {
+            dataToShow = this.dataPack.all;
+        }
+
+        /**
+         * Return the index/bucket location of the actual value.
+         * This is needed since we're using a custom X scale in order to spread the values.
+         * Example: getBucketValue(546, pop) => 13 if this value is found at index 13.
+         */
+        function getBucketValue(val, population) {
+            for (let i = 0; i < population.length; i++) {
+                if (val > population[i].xActualValue && val < population[i + 1].xActualValue) {
+                    return i;
+                }
+            }
+            return val;
+        }
```

Comment:
Nice! I'm glad this handles the case where the exact `val` is not in the dataset.